src/imp/openssl.rs

```diff
@@ -127,12 +127,19 @@ impl error::Error for Error {
         }
     }
 
-    fn cause(&self) -> Option<&error::Error> {
+    fn cause(&self) -> Option<&dyn error::Error> {
```

Comment:
You can just remove the cause implementations - they'll delegate to source anyway.